jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiResponse.java

```diff
@@ -186,9 +186,19 @@ public void sendRedirect(int code, String location) throws IOException
         {
             Response.sendRedirect(getServletRequestInfo().getRequest(), getResponse(), callback, code, location, false);
             callback.block();
+            closeOutput();
```

Comment:
Why is this necessary? The send redirect above ultimately does
```java
            response.getHeaders().put(HttpHeader.LOCATION, toRedirectURI(request, location));
            response.setStatus(code);
            response.write(true, null, callback);
```
So the last write has been done and no more content can be sent.
Is this purely to make sure an IOE is thrown even when aggregating into the HttpOutput buffer?   If so, then rather than a new method how about:
```suggestion
            getServletChannel().getHttpOutput().close();
```

Reply:
> Is this purely to make sure an IOE is thrown even when aggregating into the HttpOutput buffer?

Yes, exactly. So without this writes from a Servlet may not throw as they are just aggregated, and it only throws when it does the complete in ServletChannel. In GAE this was causing a 500 response instead of the proper redirect code to be sent if a redirect was done from a JSP.

How about I inline this method then or make it private, because I still think we need to close the writer if there is one.

Reply:
OK